### DISCUSSION TITLE
Get old filename of images uploaded to multiple image field

### DISCUSSION THREAD

**Discussion by bogdano** [pocketbase/pocketbase#5015] [Q&A] [answered]
Hello!

Is there any way to retrieve the original filename of an image uploaded to a multiple image field?

```
tmpFile, err := filesystem.NewFileFromBytes(resizedImageBytes, path.Base(imageURL))
				if err != nil {
					log.Printf("Error saving image to a temporary file: %v", err)
					continue
				}
				resizedImageBytes = nil // free up memory
				log.Printf("Image size: %f kb", float64(tmpFile.Size)/1024.0)
				// add the image to the form
				form.AddFiles("images", tmpFile)
				tmpFile = nil // free up memory
				if err := form.Submit(); err != nil {
					log.Printf("Error saving record with image: %v", err)
					continue
				}
```

Here, `tmpFile.OriginalName` will return the original name, but once the form is submitted it seems that only the new filename string is stored. 

I could create a new "images" collection and then use that collection and a one-many relation instead of a multiple image field, but that just seems messy.

Why am I doing this? My DB is populated by images downloaded from an API. When refreshing the DB data, I want to skip over the images which I've already downloaded (i.e., check the filenames of the images on my DB, and compare them to the filenames of the image in the API). So, I kind of need the original filenames in order to be able to compare. In my case, I already know that the original name is safe and unique.

**Comment by bogdano** [comment, marked as answer]
...alright I have a hacky solution here:

```
3C7B45AE-1DD8-B71B-0B7EE131C7DFC2F5.jpg becomes:

3_c7_b45_ae_1_dd8_b71_b_0_b7_ee131_c7_dfc2_f5_KPqOzo3LNg.jpg
```

So, I snooped around in the PB code to see what was inserting the extra underscores everywhere, and I found the Snakecase function in the inflector package. I then wrote two small functions to strip the filenames from urls, and to remove the unique string appended to the saved filenames by cutting off everything after the last underscore:

```
// remove file extension and everything to the left of the last '/'
func quick_strip_url(s string) string {
	filename := path.Base(s)            
	ext := path.Ext(filename)           
	return strings.TrimSuffix(filename, ext)
}

// remove everything from the right up to the first underscore from the right, from a given string
func quick_strip(s string) string {
	for i := len(s) - 1; i >= 0; i-- {
		if s[i] == '_' {
			return s[:i]
		}
	}
	return s
}
```

And then I check whether a new image already exists in the record by doing this check:

```
ImageLoop:
	for _, image := range park.Images {
		imageURL := image.URL
		// check if the image is already in the form
		for _, existingImage := range record.GetStringSlice("images") {
			if inflector.Snakecase(quick_strip_url(imageURL)) == quick_strip(existingImage) {
				log.Print("skipped")
				continue ImageLoop
			}
		}
                // code which saves the new image . . .
}
```

This works, it is a bit hacky, but may be the simplest solution without changing my schema. I'll leave this open for a couple days if anyone wants to chime in.

**Reply by softexpert** [reply]
Interesting find, thanks for sharing !
I'm curious, though, if @ganigeorgiev gives his blessing to this approach or if he has something else available / planned ...

I might add that there is an **Attributes** object that will give you the original file name, though.

```go
fsys, _ := internal.App.NewFilesystem()
defer fsys.Close()

var (
	record *models.Record{}
	attrs *blob.Attributes
	err   error
	origFileName string
)

// you get the record initialized from your query
// and then
attrs, err = fsys.Attributes(record.BaseFilesPath() + "/" + record.GetStringSlice("document_files")[0])
if err == nil{
	// here you obtain the original file name
	origFileName = attrs.Metadata["original-filename"]
	// there are other infos available, like attrs.Size, attrs.ContentType, attrs.MD5
}
```
